src/stdlib.rs

```diff
@@ -83,5 +80,14 @@ mod no_std {
                 Ok(self.inner.lock())
             }
         }
+
+        impl<T> Default for Mutex<T>
+        where
+            T: Default,
+        {
+            fn default() -> Self {
+                Mutex::new(T::default())
+            }
+        }
```

Comment:
i think the mutex should really just be removed...the eventual intention is to not use it here.